platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/system/events/UnsignedEvent.java

```diff
@@ -161,53 +169,158 @@ public BaseEventHashedData(
+        out.writeLong(eventCore.birthRound());
+        out.writeInstant(HapiUtils.asInstant(eventCore.timeCreated()));
 
         // write serialized length of transaction array first, so during the deserialization proces
         // it is possible to skip transaction array and move on to the next object
-        out.writeInt(getSerializedLength(transactions, true, false));
+        out.writeInt(PayloadUtils.getLegacyObjectSize(payloads));
         // transactions may include both system transactions and application transactions
         // so writeClassId set to true and allSameClass set to false
-        out.writeSerializableArray(transactions, true, false);
+        out.writeInt(payloads.size());
+        if (!payloads.isEmpty()) {
+            final boolean allSameClass = false;
+            out.writeBoolean(allSameClass);
+        }
+        for (final EventPayload payload : payloads) {
+            switch (payload.payload().kind()) {
+                case APPLICATION_PAYLOAD:
+                    serializeApplicationPayload(out, payload);
+                    break;
+                case STATE_SIGNATURE_PAYLOAD:
+                    serializeStateSignaturePayload(out, payload);
+                    break;
+                default:
+                    throw new IOException(
+                            "Unknown payload type: " + payload.payload().kind());
+            }
+        }
     }
 
-    @Override
-    public void deserialize(final SerializableDataInputStream in, final int version) throws IOException {
-        final TransactionConfig transactionConfig = ConfigurationHolder.getConfigData(TransactionConfig.class);
-        deserialize(in, version, transactionConfig.maxTransactionCountPerEvent());
+    private static void serializeApplicationPayload(SerializableDataOutputStream out, EventPayload payload)
+            throws IOException {
+        out.writeLong(APPLICATION_TRANSACTION_CLASS_ID);
+        out.writeInt(APPLICATION_TRANSACTION_VERSION);
+        final Bytes bytes = payload.payload().as();
+        out.writeInt((int) bytes.length());
+        bytes.writeTo(out);
     }
 
-    private void deserialize(
-            @NonNull final SerializableDataInputStream in, final int version, final int maxTransactionCount)
+    private static void serializeStateSignaturePayload(SerializableDataOutputStream out, EventPayload payload)
             throws IOException {
+        final StateSignaturePayload stateSignaturePayload = payload.payload().as();
+
+        out.writeLong(STATE_SIGNATURE_CLASS_ID);
+        out.writeInt(STATE_SIGNATURE_VERSION);
+        out.writeInt((int) stateSignaturePayload.signature().length());
+        stateSignaturePayload.signature().writeTo(out);
+
+        out.writeInt((int) stateSignaturePayload.hash().length());
+        stateSignaturePayload.hash().writeTo(out);
+
+        out.writeLong(stateSignaturePayload.round());
+        out.writeInt(Integer.MIN_VALUE); // epochHash is always null
```

Comment:
Is there a reason this is still here? I thought we were clear to get rid of it, since it isn't used anywhere

Reply:
We wanted to result in the same format for now